### DISCUSSION TITLE
Multiple sidekiq_options

### DISCUSSION THREAD

**Discussion by h0jeZvgoxFepBQ2C** [sidekiq/sidekiq#5543] [Q&A]
Hi,

what happens when I have multiple sidekiq options? 
are they overwritten or merged?

F.e. if I have a root worker class

```ruby
class RootWorker
  include Sidekiq::Worker
  sidekiq_options retry: 0, backtrace: true
end

class SubWorker < RootWorker
  sidekiq_options retry: 10, backtrace: false
end
```

What happens in the end?

Thank you for your great gem! ❤️

**Comment by mperham** [comment]
Try it in IRB and see?

**Reply by h0jeZvgoxFepBQ2C** [reply]
Actually I would prefer a confirmed statement (maybe even from you, the maintainer :)) before I upgrade around 100 workers in our production environment. Could be that only some options are overwritten and not all? wouldnt like to test this out in production.

**Reply by mperham** [reply]
Code is more trustworthy than my old memory.

```ruby
>
class RootWorker
  include Sidekiq::Worker
  sidekiq_options retry: 0, backtrace: true
end
=> {"retry"=>0, "queue"=>"default", "backtrace"=>true}
> 
class SubWorker < RootWorker
  sidekiq_options retry: 10, backtrace: false
end
=> {"retry"=>10, "queue"=>"default", "backtrace"=>false}
> RootWorker.get_sidekiq_options
=> {"retry"=>0, "queue"=>"default", "backtrace"=>true}
> SubWorker.get_sidekiq_options
=> {"retry"=>10, "queue"=>"default", "backtrace"=>false}
> 
```
